src/data_access/SudokuDAO.java

```diff
@@ -0,0 +1,153 @@
+package data_access;
+
+import okhttp3.*;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+public class SudokuDAO {
+
+    private static String encodeBoardHelper(int[][] board) {
+        StringBuilder result = new StringBuilder();
+        for (int i = 0; i < board.length; i ++) {
+            for (int j = 0; j < board[i].length; j ++ ) {
+                result.append(URLEncoder.encode(Integer.toString(board[i][j]), StandardCharsets.UTF_8));
+                if (j < board[i].length - 1) {
+                    result.append("%2C");
```

Comment:
I'm just wondering what %2C means and what the format it for the api to recognize the sudoku board

Reply:
It is just a part of the encoder that is used to make the API call to get a solution for the current board. The specific type of encoder it is, is URL.